lib/core.js

```diff
@@ -70,6 +70,9 @@ class Core extends EventEmitter2 {
     if (!this.ready) {
       throw new Error("Oak isn't ready yet, wait your turn!")
     }
+    if (_.isNull(opts.url)) {
```

Comment:
`_.isNull` will return false for `undefined` types. A better way would be checking for `!_.isString(opts.url) || !_.isFunction(opts.url)`. That way, anything else that is passed wouldn't work (Number, NaN, etc)

Reply:
ok will update